app/pages/Page.scala

```diff
@@ -16,21 +16,24 @@
 
 package pages
 
-import models.{CheckMode, Mode, NormalMode, UserAnswers}
+import models.{CheckLoopMode, CheckMode, Mode, NormalMode, UserAnswers}
 import play.api.mvc.Call
 
 import scala.language.implicitConversions
 
 trait Page {
 
   def navigate(mode: Mode, answers: UserAnswers): Call = mode match {
-    case NormalMode => navigateInNormalMode(answers)
-    case CheckMode  => navigateInCheckMode(answers)
+    case NormalMode    => navigateInNormalMode(answers)
+    case CheckMode     => navigateInCheckMode(answers)
+    case CheckLoopMode => navigateInCheckLoopMode(answers)
   }
 
   protected def navigateInNormalMode(answers: UserAnswers): Call = ???
 
   protected def navigateInCheckMode(answers: UserAnswers): Call = ???
+
+  protected def navigateInCheckLoopMode(answers: UserAnswers): Call = ???
```

Comment:
Will these stay as Not Implemented or should we just not initialise them

Reply:
Agreed we'll throw an exception instead, to be more explicit that something unexpected has happened.

Ideally we'd have a more elegant solution to modes generally to prevent the possibility of this happening altogether; one for a future refactor.